runtime/compiler/aarch64/codegen/ARM64JNILinkage.cpp

```diff
@@ -63,9 +63,44 @@ void J9::ARM64::JNILinkage::releaseVMAccess(TR::Node *callNode, TR::Register *vm
    TR_UNIMPLEMENTED();
    }
 
-void J9::ARM64::JNILinkage::acquireVMAccess(TR::Node *callNode, TR::Register *vmThreadReg)
+void J9::ARM64::JNILinkage::acquireVMAccess(TR::Node *callNode, TR::Register *vmThreadReg, TR::Register *scratchReg0, TR::Register *scratchReg1, TR::Register *scratchReg2)
    {
-   TR_UNIMPLEMENTED();
+   TR_J9VMBase *fej9 = reinterpret_cast<TR_J9VMBase *>(fe());
+
+   // Re-acquire VM access.
+   //    addimmx scratch0, vmThreadReg, #publicFlagsOffset
+   //    movzx   scratch1, constAcquireMAccessOutOfLineMask
+   //
+   // loopHead:
+   //    ldxrx   scratch2, [scratch0]
+   //    cbnzx   scratch2, reacquireVMAccessSnippet
+   //    stxrx   scratch2, scratch1, [scratch0]
+   //    cbnz    scratch2, loopHead
+   //    dmb ishld
+   // reacquireVMAccessRestartLabel:
+   //
+   generateTrg1Src1ImmInstruction(cg(), TR::InstOpCode::addimmx, callNode, scratchReg0, vmThreadReg, fej9->thisThreadGetPublicFlagsOffset());
+   generateTrg1ImmInstruction(cg(), TR::InstOpCode::movzx, callNode, scratchReg1, fej9->constAcquireVMAccessOutOfLineMask());
```

Comment:
You use `loadConstant64()` to do the same in #7996.

Reply:
I agree. Updated it.